rust/src/api.rs

```diff
@@ -195,10 +195,19 @@ impl WalletInfo {
     }
 }
 
+#[tokio::main(flavor = "current_thread")]
+pub async fn refresh_wallet_info() -> Result<WalletInfo> {
+    wallet::sync()?;
+    WalletInfo::build_wallet_info().await
+}
+
 #[tokio::main(flavor = "current_thread")]
 pub async fn run_ldk(stream: StreamSink<Event>) -> Result<()> {
     tracing::debug!("Starting ldk node");
     let background_processor = wallet::run_ldk().await?;
+    stream.add(Event::WalletInfo(
+        WalletInfo::build_wallet_info().await.ok(),
+    ));
```

Comment:
🔧 This already appears on line 212.

Reply:
This is because I want to have the initial data synchronized once, before marking the app ready.

Reply:
But you are emitting the `WalletInfo` event _twice_ before making the app ready. I think that's unintentional.

Reply:
you are right 👍 thanks!